protostar/starknet/cairo_cheatcode_syscall_handler.py

```diff
@@ -0,0 +1,204 @@
+from typing import List, cast, Optional, Any
+
+from starkware.cairo.lang.compiler.preprocessor.flow import ReferenceManager
+from starkware.cairo.lang.compiler.program import CairoHint
+from starkware.cairo.lang.vm.memory_segments import MemorySegmentManager
+from starkware.cairo.lang.vm.relocatable import RelocatableValue
+from starkware.python.utils import to_bytes
+from starkware.starknet.business_logic.execution.objects import CallType
+from starkware.starknet.business_logic.state.state_api import SyncState
+from starkware.starknet.business_logic.state.state_api_objects import BlockInfo
+from starkware.starknet.core.os.contract_address.contract_address import (
+    calculate_contract_address_from_hash,
+)
+from starkware.starknet.core.os.syscall_utils import BusinessLogicSysCallHandler
+from starkware.starknet.security.secure_hints import HintsWhitelist
+from starkware.starknet.services.api.contract_class import EntryPointType
+
+
+from .address import Address
+from .cheaters_of import cheaters_of
+
+
+class CheatableSysCallHandlerException(Exception):
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(message)
+
+
+class CheatableCairoSysCallHandler(BusinessLogicSysCallHandler):
+    def __init__(self, state: SyncState, **kwargs: Any):
+        # This field must be set before entering super constructor,
+        # because it calls the setter for the `block_info` property.
+        self.cheaters = cheaters_of(state)
```

Comment:
`CheatableCairoSysCallHandler` handler shouldn't have access to cheaters

Reply:
done